### PR TITLE
feat: cover wNEAR with tests

### LINKED ISSUE #197: wNEAR: lack of test coverage
Full test coverage is an essential process in ensuring the codebase works as intended (outlined in NEAR’s design documentation). Insufficient code coverage may lead to the arisal of unexpected issues due to changes in the underlying smart contract implementation, leading to undetected changes in functionality.

Consider updating the tests for wNEAR to include interactions with storage, `ft_transfer_call`,
`near_withdraw` and other edge cases specified in the FungibleToken standard’s documentation.

### PR DESCRIPTION
**Pull request by itegulov**
Fixes #197 

This PR migrates existing tests to workspaces and add new ones to cover storage functions, `ft_transfer_call` and wNEAR-specific functions.

I will do another pass tomorrow to make sure I haven't missed anything.

### CODE REVIEW

File: w-near/Cargo.toml [hunk 1]
```diff
@@ -5,14 +5,18 @@ authors = ["Near Inc <hello@nearprotocol.com>", "Vincent <vincent@decentralisede
...
 
 [lib]
-crate-type = ["cdylib", "rlib"]
+crate-type = ["cdylib"]
 
 [dependencies]
-near-sdk = "3.1.0"
-near-contract-standards = "3.1.0"
+near-sdk = "4.0.0-pre.6"
+near-contract-standards = "4.0.0-pre.6"
```
Comment (austinabell) [inline w-near/Cargo.toml:12]: Note to whoever manages this contract that the version is being updated.

Prob worth using pre.7 to get the events added in with the ft?

File: w-near/test-w-near-defi/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    fn value_please(&self, amount_to_return: String) -> PromiseOrValue<U128>;
+}
+
+#[near_bindgen]
+impl DeFi {
+    #[init]
+    pub fn new(fungible_token_account_id: AccountId) -> Self {
+        require!(!env::state_exists(), "Already initialized");
+        Self {
+            fungible_token_account_id: fungible_token_account_id.into(),
```
Comment (austinabell) [inline w-near/test-w-near-defi/src/lib.rs:41]:
```suggestion
            fungible_token_account_id,
```

shouldn't be needed, should it? I know it's just used for testing but figured I'd mention

**Review by austinabell — COMMENTED** [review]
Looks generally good to me. Will have another scan through tomorrow but still not sure who manages/owns the wNEAR contract to upgrade

File: w-near/Cargo.toml (hunk 1 above)
Comment (itegulov) [reply w-near/Cargo.toml]: Yeah I do not see a reason not to upgrade. Done!
